### PR TITLE
Update FFI API with key certification and signature customization functions.

### LINKED ISSUE #1085: Add feature to certify other keys.
## Description

It should be added on library, FFI and CLI layer.
For CLI it may be implemented as `rnpkeys --certify <keyname> --signer <signer>` command (without signer default key may be used). In addition trust level parameters should be added.

### PR DESCRIPTION
**Pull request by ni4**
This PR update FFI API with a number of functions for signature customization or information retrieval. Also it adds functions to create self-certifications and other key certification creation.

Fixes #1085

### CODE REVIEW

File: include/rnp/rnp.h [hunk 1]
```diff
@@ -1728,9 +1902,162 @@ RNP_API rnp_result_t rnp_signature_get_creation(rnp_signature_handle_t sig, uint
...
+ * @return RNP_SUCCESS or error code if failed.
+ */
+RNP_API rnp_result_t rnp_signature_get_preferred_alg_count(rnp_signature_handle_t sig,
+                                                           size_t *               count);
+
+/**
+ * @brief Get preferred symmetric algorithm from the preferences, specified in the signature.
+ *
+ * @param sig signature handle, cannot be NULL.
+ * @param idx index in the list. Number of availabke items could be obtained via the
```
Comment (desvxx) [inline include/rnp/rnp.h:1938]: typo (available)

File: include/rnp/rnp.h [hunk 2]
```diff
@@ -1728,9 +1902,162 @@ RNP_API rnp_result_t rnp_signature_get_creation(rnp_signature_handle_t sig, uint
...
+ * @return RNP_SUCCESS or error code if failed.
+ */
+RNP_API rnp_result_t rnp_signature_get_preferred_hash_count(rnp_signature_handle_t sig,
+                                                            size_t *               count);
+
+/**
+ * @brief Get preferred hash algorithm from the preferences, specified in the signature.
+ *
+ * @param sig signature handle, cannot be NULL.
+ * @param idx index in the list. Number of availabke items could be obtained via the
```
Comment (desvxx) [inline include/rnp/rnp.h:1964]: typo

**Review by desvxx — APPROVED** [review]

File: include/rnp/rnp.h (hunk 1 above)
Comment (ni4) [reply include/rnp/rnp.h]: Fixed (here, below and issued PR #2272) , thanks.

**Review by maxirmx — APPROVED** [review]
LGTM

**Comment by ni4** [comment]
Let's merge it, having 2 approvals and green CI.

**Pull request merged**